### DISCUSSION TITLE
Way to update scrypted without removing and replacing container. Way to do it through UI?

### DISCUSSION THREAD

**Discussion by justinshuls** [koush/scrypted#192] [Ideas]
Way to update scrypted without removing and replacing container. Way to do it through UI?

**Comment by miwagner1** [comment]
Installing scrypted natively will let you do it through the UI.

Containers are disposable in docker, which is why you set volumes for persistent data. The docker way is to pull a new image and recreate the service by taking down the container and replacing it with a new one using the new image you just pulled as it’s base. I highly recommend using docker compose to make life much easer.


If you want automatic updates with docker, use watchtower or set up a cron job or something similar.

**Comment by koush** [comment]
There's no way to do a docker update from outside of docker (within scrypted).

**Comment by cheme75** [comment]
Fwiw, my docker install, added docker-compose and put c:\docker\docker\resources\bin (location of docker-compose) in the environment path so it can be run anywhere. In win, I update containers with docker-compose pull, run from cmd prompt after cd to folder with the docker-compose.yml file. I finally found out via Wyze-bridge discussion that it’s easier to use docker-compose and setup the yml file for each container in its own folder. Then just cd to each folder,  then run, docker-compose pull, to update, then run docker-compose up -d, to run it. Then run docker image prune to remove the old image. 

Fwiw, I think the support for scrypted is about the worst I’ve come across so far. Wyze-bridge has an active developer and several users who offer really good support.

**Reply by miwagner1** [reply]
Docker compose does what I said. You pull a new image, docker compose up -d will stop and destroy the current running container and spin up a new one using the new image that was pulled. It “Updates” the container by making a new one.

The user asked if there was a way without removing and replacing the container. That answer is still no, even with docker compose / swarm / Kubernetes.

Now it is possible like with https://hub.docker.com/r/linuxserver/plex
But that is rare and “ Most of our images are static, versioned, and require an image update and container recreation to update the app inside.”

**Reply by cheme75** [reply]
ok, was just thinking using docker-compose via cmd line is outside of docker, but ya, I see the remove/replace is no different, really.  Meanwhile I just got scrypted running but not clear why my browsers won't connect, opened port 10443 but still nothing.
